numpy/core/_add_newdocs.py

```diff
@@ -1084,12 +1084,25 @@
 
     Examples
     --------
-    >>> x = np.arange(6).reshape(2,3)
-    >>> np.ascontiguousarray(x, dtype=np.float32)
-    array([[0., 1., 2.],
-           [3., 4., 5.]], dtype=float32)
+    Try to change memory layout of ndarray from C-contiguous to C-contiguous::
```

Comment:
```suggestion
    Try to change memory layout of ndarray from C-contiguous to C-contiguous:
    
```
I think the typical pattern is a blank line and no `::`.

Reply:
I might suggest a wording like:  `If we create a C-contiguous ``ascontiguousarray`` will return it without:`

Could even swap the order for the C-contiguous, so we always lead with the "copy made" example (seems a bit more relevant to me).

Reply:
I thought numpy is using double-colons to introduce code blocks:

https://github.com/numpy/numpy/blob/main/numpy/core/_add_newdocs.py#L1569
https://github.com/numpy/numpy/blob/main/numpy/core/_add_newdocs.py#L1705
https://github.com/numpy/numpy/blob/main/numpy/core/_add_newdocs.py#L1717
etc.

At least according to sphinx documentation `Code blocks are introduced by a double-colon and are indented`. Anyways, if you prefer to use single-colon or no colon at all let me know.

Reply:
That is true in general, but our docstrings are processed by numpydoc before it goes to sphinx. You can easily find some examples in the code, or look at https://numpydoc.readthedocs.io/en/latest/format.html.

Reply:
Thanks, yes. I see examples using single-colon and no indentation and also examples using double-colon two-space indentation. I will adhere to the former.

Reply:
It is done that way because

1. The documentation looks better in a terminal.
2. The documentation is easier to format.

Reply:
I just wonder why different parts of documentation use different styles (double colon/two spaces vs single colon/no indentation). Historic reasons?

Reply:
> Historic reasons?

Probably oversight on the reviewers part :) I suspect that numpydoc just ignores the colons, not sure what it does with the indentation.

@rossbar @jarrodmillman Can you expand on this?

Reply:
It's more complicated than it should be...

My understanding is that in the example section, single quotes (without indentation) is preferable because numpydoc will automatically grab code that starts with `>>>` and highlight it properly.

*Outside* of the `Examples` section, yes - you should use `::` to specify a code block to get proper formatting + syntax highlighting, etc. In other words:

```rst

Here's my code example in the summary part of the docstring::

    >>> print("foo")
    foo

Examples
--------
Here's my docstring example:

>>> print("bar")
bar
```